packages/framework/src/resources/workflow/discover-channel-step-factory.ts

```diff
@@ -32,14 +40,14 @@ export async function discoverChannelStepFactory(
       },
       resolve: resolve as (controls: Record<string, unknown>) => Awaitable<Record<string, unknown>>,
       code: resolve.toString(),
-      options,
+      options: options as StepOptions<FromSchema<Schema>, FromSchema<Schema>>,
       providers: [],
     };
 
     await discoverStep(targetWorkflow, stepId, step);
 
     if (Object.keys(options.providers || {}).length > 0) {
-      discoverProviders(step, type as ChannelStepEnum, options.providers || {});
+      await discoverProviders(step, type as ChannelStepEnum, options.providers || {});
```

Comment:
This was missing before and could result in providers not being discovered inside the async event loop.